browser_use/agent/service.py

```diff
@@ -163,14 +163,15 @@ def __init__(
 		enable_memory: bool = True,
 		memory_config: MemoryConfig | None = None,
 		source: str | None = None,
+		search_engine: str = 'google',  # Moved to end to avoid breaking existing positional arguments
 	):
 		if page_extraction_llm is None:
 			page_extraction_llm = llm
 
 		# Core components
 		self.task = task
 		self.llm = llm
-		self.controller = controller
+		self.controller = controller or Controller(search_engine=search_engine)
```

Comment:
I don't want the Controller to manage action-specific configuration at the moment.

We haven't decided how we want to design configuration for actions yet. Could you change it to use an environment variable instead for now, then we could merge it faster.

e.g. in side your action just check for a `os.getenv('BROWSERUSE_SEARCH_ENGINE', 'google').lower().strip().split('.')[0]` and `assert search_engine in search_engines`